genie-core/src/main/java/com/netflix/genie/core/jobs/JobLauncher.java

```diff
@@ -85,45 +80,35 @@ public JobLauncher(
         this.registry = registry;
-        this.submitTimer = this.registry.timer("genie.jobs.submit.timer");
+        this.submitTimerId = this.registry.createId("genie.jobs.submit.timer");
     }
 
     /**
      * Starts the job setup and launch process once the thread is activated.
      */
     @Override
     public void run() {
-        this.submitTimer.record(() -> {
-                // TODO: Compress the timer into a single ID tagged by exceptions
-                try {
-                    this.jobSubmitterService.submitJob(
-                        this.jobRequest,
-                        this.cluster,
-                        this.command,
-                        this.applications,
-                        this.memory
-                    );
-                } catch (final GenieException e) {
-                    log.error("Unable to submit job due to exception: {}", e.getMessage(), e);
-                    if (e instanceof GenieBadRequestException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_BAD_REQUEST_RATE).increment();
-                    } else if (e instanceof GenieConflictException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_CONFLICT_RATE).increment();
-                    } else if (e instanceof GenieNotFoundException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_NOT_FOUND_RATE).increment();
-                    } else if (e instanceof GeniePreconditionException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_PRECONDITION_RATE).increment();
-                    } else if (e instanceof GenieServerException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_SERVER_RATE).increment();
-                    } else if (e instanceof GenieServerUnavailableException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_SERVER_UNAVAILABLE_RATE).increment();
-                    } else if (e instanceof GenieTimeoutException) {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_TIMEOUT_RATE).increment();
-                    } else {
-                        this.registry.counter(MetricsConstants.GENIE_EXCEPTIONS_OTHER_RATE).increment();
-                    }
-                }
-            }
-        );
+        final long start = System.nanoTime();
+        final Map<String, String> tags = MetricsConstants.newSuccessTagsMap();
+        try {
+            this.jobSubmitterService.submitJob(
+                this.jobRequest,
+                this.cluster,
+                this.command,
+                this.applications,
+                this.memory
+            );
+        } catch (final GenieException e) {
+            log.error("Unable to submit job due to exception: {}", e.getMessage(), e);
+            MetricsConstants.addFailureTagsWithException(tags, e);
+        } catch (Throwable t) {
```

Comment:
Wrapping would change the behavior, e.g. if it's a `RuntimeException` it would have propagated before, and it should be propagated now, I'm just catching to tag the timer and then letting it go as it was before.